contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+
+/**
+ * @dev A child contract which will be deployed by the ERC20MultiDelegate utility contract
+ * This is a proxy delegator contract to vote given delegatee on behalf of original delegator
+ */
+contract ERC20ProxyDelegator {
+    constructor(ERC20Votes _token, address _delegatee) {
```

Comment:
Weren't we going to have this selfdestruct as well, to save gas?

Reply:
after our last discussion as I remember the conclusion was; we do not need it.

Reply:
I wonder if it would be worth the extra gas to store the token, delegatee, and msg.sender values as immutable public variables?

Reply:
| depositMulti | none | public | public immutable |
| ---- | --- | --- | --- |
| min | 448520 | 766173 | 631925 | 
| max | 864048 | 1499405 | 1230909 |

Reply:
An extra 50%! Let's leave it out, then.

Reply:
Worth investigating whether writing this in Yul would save a lot of gas, too.

Reply:
keeping this conversation open for future improvements